AWSAppSyncClient/Internal/AWSPerformMutationOperation.swift

```diff
@@ -69,35 +102,54 @@ final class AWSPerformMutationOperation<Mutation: GraphQLMutation>: Asynchronous
             }
         }
     }
-
-    // MARK: Operation
-
-    override func start() {
-        if isCancelled {
-            state = .finished
-            return
+    
+    private func scheduleRetry() {
+        AppSyncLog.debug("Scheduling mutation retry for in-memory mutation.")
+        mutationRetryHelper = AWSMutationRetryHelper(retryAttemptNumber: currentAttemptNumber) {
```

Comment:
I believe this is going to result in a data race where the signal callback is being invoked on a different queue than `self` is being assigned.

Reply:
I will investigate this and follow up.

Reply:
I looked and I think I understand how it could result in data race(less likely currently due to the wait from the timer but still CAN happen). I will update the code to ensure that we do not run into that scenario. Great catch!